src/main/java/jsr223/nativeshell/shell/Shell.java

```diff
@@ -1,15 +1,39 @@
 package jsr223.nativeshell.shell;
 
+import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
 
 import javax.script.ScriptEngineFactory;
+import javax.script.ScriptException;
 
 import jsr223.nativeshell.NativeShell;
 
 
 public class Shell implements NativeShell {
+
+    private static final String SHEBANG_PREFIX = "#!";
+
     @Override
-    public ProcessBuilder createProcess(File commandAsFile) {
+    public ProcessBuilder createProcess(File commandAsFile) throws ScriptException {
+        try (FileReader fileReader = new FileReader(commandAsFile);
+             BufferedReader reader = new BufferedReader(fileReader)) {
+            final String shebangLine = reader.readLine();
+            if( shebangLine.startsWith(SHEBANG_PREFIX) && shebangLine.length() > SHEBANG_PREFIX.length()) {
+                final String pathToInterpreter = shebangLine.substring(SHEBANG_PREFIX.length());
+                final File interpreter = new File(pathToInterpreter);
+                if (!interpreter.exists() || interpreter.isDirectory()) {
```

Comment:
could you differentiate the error messages ?
Also, a interpreter.canExecute() makes sense

Reply:
ok